compiler/visq/visqlib/DumpFP32FM.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script that dumps FM of fp32 model
+# NOTE This script runs on dalgona
+
+import numpy as np
+
+from pathlib import Path
+from Util import to_filename
+
+
+# Dump fp32 model's intermediate FM data and their names
+#
+# Before
+# self._dir/
+#
+# After
+# self._dir/
+#  tensors.txt
+#  <TENSOR_NAME>.npy
+# NOTE TENSOR_NAME is transformed by to_filename
+class DumpFP32FM(object):
```

Comment:
Just curiosity, what is this `object` keyword for?

Reply:
(optional) And if there is an interface for dalgona, it would be helpful to write a dalgona analysis script.

Reply:
(optional2) It is not related this PR but we can introduce an option that receives python script and the class name to analyze. Then, users manages multiple simple classes by just a one python script.

For example,
```bash
$ cat DumpFM.py
class DumpFP32FM(object):
..
class DumpFakeQuantFM(object):
..


```

Reply:
> Just curiosity, what is this object keyword for?

`object` is unnecessary. Thanks for finding.

> (optional) And if there is an interface for dalgona, it would be helpful to write a dalgona analysis script.

DumpFP32FM.py and DumpFakeQuantFM,py are dalgona analysis scripts.

> (optional2) It is not related this PR but we can introduce an option that receives python script and the class name to analyze. Then, users manages multiple simple classes by just a one python script.

Dalgona analysis should have the class with the name same with the file name. Ex: DumpFP32FM class in DumpFP32FM.py.

Reply:
@jinevening 
> DumpFP32FM.py and DumpFakeQuantFM,py are dalgona analysis scripts.

Right. I think if we can force dalgona script writers to implement the things that should be implemented by interface.

For example,
```python
class DumpFP32FM(IDalgona):
  # if `StartAnalysis` is missing, throw an error
```

> Dalgona analysis should have the class with the name same with the file name. Ex: DumpFP32FM class in DumpFP32FM.py.

Right. I just wanted to check if we could consider changing the policy where one file can have multiple analyzing classes. But current policy looks good as well.

Reply:
> I think if we can force dalgona script writers to implement the things that should be implemented by interface.

At least for now, I think `StartAnalysis` (or any other method) does not have to be implemented. We can think of such an interface when necessary.

> I just wanted to check if we could consider changing the policy where one file can have multiple analyzing classes. But current policy looks good as well.

Hmm.. This may change the interface of `dalgona`. I'm not sure we can do this easily. Let's think about it later :)